libsrc/atari2600/ctype.s

```diff
@@ -1,162 +1,5 @@
-;
-; Ullrich von Bassewitz, 2003-10-10
-;
 ; Character specification table.
 ;
+; uses the "common" definition
```

Comment:
I thought this "include pattern" could be replaced by that "vpath shadowing pattern". Could you please elaborate why it doesn't work here (and for the other files using the same pattern)?

Reply:
Because the 'ctype_common.inc' is/was different from the the 'ctype_console.inc'. Those both files match to most targets and I prefer a central definition. I thought you, being an Apple 2 guy, prefer the Apple 2 definition to be the default one... ;-)

Reply:
Maybe I'm just slow on take up but I still don't get it :-(

In https://github.com/cc65/cc65/blob/master/src/common/target.c#L183-L218 we have a table stating which target uses which char map. I would presume that:

* All targets listed there with `CTNone` do NOT have a ctype.s file in their libsrc directory - as they just all "fall back" to the ASCII ctype.s in libsrc/common.

* All targets which are listed there with some other `CT<...>` else DO have a ctype.s file in their libsrc directory (or in the libsrc/cbm directory).

So the only scenario where there's a need for including a file would be "secondary" targets like the telestrat based on the atmos - and there one includes the "primary" ctype.s, but not an .inc file.

Reply:
There is a difference between translation and classification.
The ´CTNone´ applies to all targets with the same translation.
These targets however, differ in classification of the characters. Some of them support only 128 characters, leaving the rest 'unclassified' (ctype_common.inc) while others having inverted characters when the high bit is set (former ctype_console.inc now ctype.s). To those characters the classification is independent from the most significant bit.
For the first group of machines I like to share the definition rather than copy it to each target (which was the state before my PR), this is done be including the common definition.

Reply:
Thanks for the elaborate explanation :-)